### PR TITLE
[22027] Remove Orchestrator from Front-end framework app

### PR DESCRIPTION
**Pull request by JesusPoderoso**
This PR removes the orchestrator usage from the front-end framework app, as it is deployed separately in another python node.
It performs all communication pipeline through API REST.

This PR is on top of (and must be merged after):
- #15 

Related PRs:
- https://github.com/eProsima/SustainML-Library/pull/58

### CODE REVIEW

File: include/sustainml/Engine.h [hunk 1]
```diff
@@ -24,18 +24,17 @@
```
Comment (Mario-DL) [inline include/sustainml/Engine.h:24]: We could take the chance to rename this header to `Engine.hpp`

File: include/sustainml/Engine.h [hunk 2]
```diff
@@ -153,20 +148,19 @@ public slots:
...
 
     size_t split_string(
             const std::string& string,
-            std::vector<std::string>& string_set,
+            QJsonArray& string_array,
             char delimeter);
 
-
-    sustainml::orchestrator::OrchestratorNode* orchestrator;
+    QNetworkAccessManager* user_input_request_;
```
Comment (Mario-DL) [inline include/sustainml/Engine.h:162]: I dont see where this member pointer gets freed in termination.

File: include/sustainml/Engine.h [hunk 3]
```diff
@@ -153,20 +148,19 @@ public slots:
...
+    std::array<QNetworkAccessManager*, static_cast<size_t>(sustainml::NodeID::MAX)> node_responses_;
```
Comment (Mario-DL) [inline include/sustainml/Engine.h:163]: Same comment, I think that these pointers are not being deleted when the application exits

**Review by Mario-DL — CHANGES_REQUESTED** [review]

File: include/sustainml/Engine.h [hunk 4]
```diff
@@ -137,14 +117,29 @@ public slots:
...
 
+    const QString server_url_ = "http://127.0.0.1:5001";
+
     QString get_name_from_node_id(
             const sustainml::NodeID& id);
 
-    QString get_task_from_data(
-            const sustainml::NodeID& id,
-            void* data);
+    sustainml::NodeID get_node_id_from_name(
```
Comment (Mario-DL) [inline include/sustainml/Engine.h:135]: We would need to add a brief doxygen in the new methods

File: include/sustainml/Engine.h (hunk 1 above)
Comment (JesusPoderoso) [reply include/sustainml/Engine.h:24]: I've done it in a [follow up PR](https://github.com/eProsima/SustainML-Framework/pull/18) ([this commit](https://github.com/eProsima/SustainML-Framework/pull/18/commits/44db0c3f0f627d7a5dcad64e42481b21f184f9a1))

File: include/sustainml/Engine.h (hunk 4 above)
Comment (JesusPoderoso) [reply include/sustainml/Engine.h:135]: I've done it in a [follow up PR](https://github.com/eProsima/SustainML-Framework/pull/18) ([this commit](https://github.com/eProsima/SustainML-Framework/pull/18/commits/46841c1b64ac1424ddf2d65af0e073bad80ba0ee)), as well as moving them to an Utils class.

**Review by Mario-DL — APPROVED** [review]
LGTM
We may have a race when accessing `node_responses_` array, but IDK right now if `launch_task()` can happen at the same time of destruction. We could address that in a future PR. Also, introducing a ci in this repo would be needed

**Pull request merged**